### DISCUSSION TITLE
No data shown on grafana dashboard for milvus when deploy monitor with k8s offline

### DISCUSSION THREAD

**Discussion by xiaojuan8** [milvus-io/milvus#15968] [Q&A and General discussion] [answered]
hi, I deployed milvus monitor service with k8s offline, ref: https://milvus.io/cn/docs/v2.0.0/install_offline-helm.md, but no data shown on grafana dashboard.
<img width="1152" alt="image" src="https://user-images.githubusercontent.com/74544828/157584397-647910c7-1272-470d-b649-e7f450ed21e0.png">
and the command for getting milvus_manifest.yaml is as follows:
helm template --set cluster.enabled=true --set metrics.serviceMonitor.enabled=true milvus milvus/milvus -n milvus > xxxxxxmilvus_manifest.yaml

my monitor service is like this:
![image](https://user-images.githubusercontent.com/74544828/157584587-d5237780-ad8f-4909-b28b-f26762ec6d8e.png)

but no data show on grafana:
![image](https://user-images.githubusercontent.com/74544828/157584642-a2653a71-239b-480b-aa92-2dd7c11138cd.png)

**Comment by xiaojuan8** [comment]
hi, could you help me on this issue? @LoveEachDay

**Comment by xiaojuan8** [comment]
Does anyone have any clues about this issue?

**Comment by LoveEachDay** [comment]
Would you provide the `servicemonitor.yaml` ? You should keep the servicemonitor and milvus cluster in the same namespace?  or set the namespace selector.

And you can verify whether your setting is correct by checking the prometheus scrape target.

**Reply by xiaojuan8** [reply]
👌thanks， I try to check those infos

and servicemonitor.yaml is listed as :

cat ServiceMonitor.yaml
-------------------------------------------------------
apiVersion: monitoring.coreos.com/v1
kind: ServiceMonitor
metadata:
  name: milvus
  namespace: monitoring
  labels:
    helm.sh/chart: milvus-3.0.6
    app.kubernetes.io/name: milvus
    app.kubernetes.io/instance: milvus
    app.kubernetes.io/version: "2.0.0"
    app.kubernetes.io/managed-by: Helm
spec:
  endpoints:
    - honorLabels: true
      interval: 30s
      scrapeTimeout: 10s
      path: /metrics
      port: metrics
  namespaceSelector:
    matchNames:
      - milvus
  selector:
    matchLabels:
      app.kubernetes.io/name: milvus
      app.kubernetes.io/instance: milvus
  targetLabels:
    - app.kubernetes.io/name
    - app.kubernetes.io/instance
    - component

**Comment by LoveEachDay** [comment, marked as answer]
This `servicemonitor` looks good to me.

Can you check the service inside the `milvus` namespace?
`kubectl get svc -n milvus -l app.kubernetes.io/name=milvus,app.kubernetes.io/instance=milvus`

And make sure those services all have endpoints.

And also you can access the prometheus web ui, and check the `targets` under the `status` menu. If nothing is wrong, you can see the targets like this:
![image](https://user-images.githubusercontent.com/1573213/157631476-4da03848-d81b-489c-a35c-52fa724af6ae.png)

**Reply by xiaojuan8** [reply]
Yes. Actually, something wrong with Prothemeus, thanks!
We consider to solve it
